runtime/tests/decimal_conv.c

```diff
@@ -51,6 +48,36 @@ void testDecToFloat() {
     validatDecToFloat("-1.7976931348623156e+309", -INFINITY);
 }
 
+void validateFloatToDec(double val, const char *decStr, PanicCode code) {
+    validate(_bal_decimal_from_float(val), decStr, code);
+}
+
+void testFloatToDec() {
+    validateFloatToDec(0.0, "0", 0);
+    validateFloatToDec(1.0, "1.0", 0);
```

Comment:
The output should not have `.0`.

Reply:
@jclark, in [subset](http://speleotrove.com/decimal/dax3274.html), it says,
```
If the coefficient in a result has the value zero, then the sign is set to 0 and (unless the operation is quantize) the exponent is set to 0.
```

So as I understood, we can trim the `.0` only if the value is zero.
I checked this, by generating the decString giving `1.0` in the decNumber library which is compiled for subset and I got `1.0` as the output.

Also the output of `dtoa` function is 1.0, if the input double is 1.0.

Reply:
We decided that the correct output for float to decimal was to have the minimum number digits needed to round-trip the value.  Therefore there should never a trailing `.0`.

Reply:
Changed